text/0000-add-limits-trait.md

```diff
@@ -0,0 +1,133 @@
+- Feature Name: Limits trait for the rust types
+- Start Date: 2017-12-20
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+This is an RFC to add a universal trait for the type limits.
+
+# Motivation
+The motivation is quite simple: [make an ability to accept template types with with limits by requiring a trait](https://stackoverflow.com/questions/47904954/rust-finding-the-maximum-allowable-value-for-generic-data-type-t) 
+so that it simplifies and generalizes the code. Another motivation is that we have all that `max_value()` and `min_value()` implemented as
+usual methods of a type implementation, generalizing this to a trait makes the code simplier and avoids duplicating code. Also, looking at
+C++ template of `std::numeric_limits` tells us we must have this thing too because it is easier to use.
+
+# Detailed design
+The design is quite simple: put everything related to the limits what can be generalized into a separate trait in the standard library:
+
+```rust
+trait Limits<T> {
```

Comment:
This should probably be `<T = Self>` if we allow a type parameter `T` there to make the use more ergonomic.
See [`std::cmp::PartialEq`](https://doc.rust-lang.org/nightly/std/cmp/trait.PartialEq.html) and (a lot of) friends for precedent.

Reply:
Good notice!